lms/resources/_js_config/__init__.py

```diff
@@ -231,20 +232,8 @@ def enable_error_dialog_mode(self, error_code, error_details=None, message=None)
             )
         )
 
-    def enable_dashboard_mode(self, assignment):
-        self._config.update(
-            {
-                "mode": JSConfig.Mode.DASHBOARD,
-                "dashboard": {
-                    "assignment": {"title": assignment.title},
-                    "assignmentStatsApi": {
-                        "path": self._request.route_path(
-                            "api.assignment.stats", id_=assignment.id
-                        ),
-                    },
-                },
-            }
-        )
+    def enable_dashboard_mode(self, config: DashboardConfig):
```

Comment:
Changed this method's approach. Instead of setting the config itself it just puts it in the right place.

The view module knows everything about dashboard this way.